### ISSUE TITLE
Yarn produces scarbled output when running on windows

### ISSUE THREAD

**Issue by doberkofler** [yarnpkg/yarn#2114]
**Do you want to request a *feature* or report a *bug*?**
possible bug

**What is the current behavior?**
a somehow scrambled output is generated
```cmd
D:\MyDev\ljs_app\trunk\periscope\examples\yarn\hanging>yarn
yarn install v0.17.10
info No lockfile found.
[1/4] Resolving packages...
warning chromedriver > npmconf@2.1.2: this package has been reintegrated into npm and is now out of date with respect to npm
warning gulp-add-src > vinyl-fs > glob-stream > minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning gulp-add-src > through2 > xtend > object-keys@0.4.0:
warning karma-phantomjs-launcher > phantomjs-prebuilt > request > node-uuid@1.4.7: use uuid module instead
warning gulp-add-src > vinyl-fs > glob-watcher > gaze > globule > minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning gulp-add-src > vinyl-fs > glob-watcher > gaze > globule > glob > graceful-fs@1.2.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Pl
ease update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
[2/4] Fetching packages...
warning fsevents@1.0.15: The platform "win32" is incompatible with this module.
info "fsevents@1.0.15" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning Unmet peer dependency "stylus@>=0.52.4".
[4/4] Building fresh packages...
[1/8] ⠄ fsevents: not ok
[8/8] ⠄ phantomjs-prebuilt
[7/8] ⡀ pngquant-bin
[-/8] ⡀ waiting...
error Error running install script for optional dependency: "D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_modules\\fsevents: Command fail
ed.\nExit code: 1\nCommand: C:\\Windows\\system32\\cmd.exe\nArguments: /d /s /c node-pre-gyp install --fallback-to-build\nDirectory: D:\\MyDev\\ljs_app\\trunk\\
periscope\\examples\\yarn\\hanging\\node_modules\\fsevents\nOutput:\nnode-pre-gyp info it worked if it ends with ok\nnode-pre-gyp info using node-pre-gyp@0.6.29
\nnode-pre-gyp info using node@6.9.1 | win32 | x64\nnode-pre-gyp info check checked for \"D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_mo
dules\\fsevents\\lib\\binding\\Release\\node-v48-win32-x64\\fse.node\" (not found)\nnode-pre-gyp http GET https://fsevents-binaries.s3-us-west-2.amazonaws.com/v
1.0.15/fse-v1.0.15-node-v48-win32-x64.tar.gz\nnode-pre-gyp http 404 https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.0.15/fse-v1.0.15-node-v48-win32-x64.
tar.gz\nnode-pre-gyp ERR! Tried to download: https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.0.15/fse-v1.0.15-node-v48-win32-x64.tar.gz \nnode-pre-gyp E
RR! Pre-built binaries not found for fsevents@1.0.15 and node@6.9.1 (node-v48 ABI) (falling back to source compile with node-gyp) \nnode-pre-gyp http Pre-built
binary not available for your system, looked for https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.0.15/fse-v1.0.15-node-v48-win32-x64.tar.gz \n\r\nD:\\My
Dev\\ljs_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_modules\\fsevents>if not defined npm_config_node_gyp (node \"C:\\Users\\doberkofler.LBITS\\AppData
\\Roaming\\npm\\node_modules\\yarn\\bin\\node-gyp-bin\\\\..\\..\\node_modules\\node-gyp\\bin\\node-gyp.js\" clean )  else (node  clean ) \r\ngyp info it worked
if it ends with ok\ngyp info using node-gyp@3.4.0\ngyp info using node@6.9.1 | win32 | x64\ngyp info ok \n\r\nD:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\ya
rn\\hanging\\node_modules\\fsevents>if not defined npm_config_node_gyp (node \"C:\\Users\\doberkofler.LBITS\\AppData\\Roaming\\npm\\node_modules\\yarn\\bin\\nod
e-gyp-bin\\\\..\\..\\node_modules\\node-gyp\\bin\\node-gyp.js\" configure --fallback-to-build --module=D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\yarn\\han
ging\\node_modules\\fsevents\\lib\\binding\\Release\\node-v48-win32-x64\\fse.node --module_name=fse --module_path=D:\\MyDev\\ljs_app\\trunk\\periscope\\examples
\\yarn\\hanging\\node_modules\\fsevents\\lib\\binding\\Release\\node-v48-win32-x64 )  else (node  configure --fallback-to-build --module=D:\\MyDev\\ljs_app\\tru
nk\\periscope\\examples\\yarn\\hanging\\node_modules\\fsevents\\lib\\binding\\Release\\node-v48-win32-x64\\fse.node --module_name=fse --module_path=D:\\MyDev\\l
js_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_modules\\fsevents\\lib\\binding\\Release\\node-v48-win32-x64 ) \r\ngyp info it worked if it ends with ok
\ngyp info using node-gyp@3.4.0\ngyp info using node@6.9.1 | win32 | x64\ngyp info spawn C:\\Python27\\python.EXE\ngyp info spawn args [ 'C:\\\\Users\\\\doberko
fler.LBITS\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\yarn\\\\node_modules\\\\node-gyp\\\\gyp\\\\gyp_main.py',\ngyp info spawn args   'binding.gyp',\ngyp i
nfo spawn args   '-f',\ngyp info spawn args   'msvs',\ngyp info spawn args   '-G',\ngyp info spawn args   'msvs_version=auto',\ngyp info spawn args   '-I',\ngyp
 info spawn args   'D:\\\\MyDev\\\\ljs_app\\\\trunk\\\\periscope\\\\examples\\\\yarn\\\\hanging\\\\node_modules\\\\fsevents\\\\build\\\\config.gypi',\ngyp info
spawn args   '-I',\ngyp info spawn args   'C:\\\\Users\\\\doberkofler.LBITS\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\yarn\\\\node_modules\\\\node-gyp\\\\
addon.gypi',\ngyp info spawn args   '-I',\ngyp info spawn args   'C:\\\\Users\\\\doberkofler.LBITS\\\\.node-gyp\\\\6.9.1\\\\include\\\\node\\\\common.gypi',\ngy
p info spawn args   '-Dlibrary=shared_library',\ngyp info spawn args   '-Dvisibility=default',\ngyp info spawn args   '-Dnode_root_dir=C:\\\\Users\\\\doberkofle
r.LBITS\\\\.node-gyp\\\\6.9.1',\ngyp info spawn args   '-Dnode_gyp_dir=C:\\\\Users\\\\doberkofler.LBITS\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\yarn\\\\
node_modules\\\\node-gyp',\ngyp info spawn args   '-Dnode_lib_file=node.lib',\ngyp info spawn args   '-Dmodule_root_dir=D:\\\\MyDev\\\\ljs_app\\\\trunk\\\\peris
cope\\\\examples\\\\yarn\\\\hanging\\\\node_modules\\\\fsevents',\ngyp info spawn args   '--depth=.',\ngyp info spawn args   '--no-parallel',\ngyp info spawn ar
gs   '--generator-output',\ngyp info spawn args   'D:\\\\MyDev\\\\ljs_app\\\\trunk\\\\periscope\\\\examples\\\\yarn\\\\hanging\\\\node_modules\\\\fsevents\\\\bu
ild',\ngyp info spawn args   '-Goutput_dir=.' ]\nTraceback (most recent call last):\r\n  File \"C:\\Users\\doberkofler.LBITS\\AppData\\Roaming\\npm\\node_module
s\\yarn\\node_modules\\node-gyp\\gyp\\gyp_main.py\", line 16, in <module>\r\n    sys.exit(gyp.script_main())\r\n  File \"C:\\Users\\doberkofler.LBITS\\AppData\\
Roaming\\npm\\node_modules\\yarn\\node_modules\\node-gyp\\gyp\\pylib\\gyp\\__init__.py\", line 545, in script_main\r\n    return main(sys.argv[1:])\r\n  File \"
C:\\Users\\doberkofler.LBITS\\AppData\\Roaming\\npm\\node_modules\\yarn\\node_modules\\node-gyp\\gyp\\pylib\\gyp\\__init__.py\", line 538, in main\r\n    return
 gyp_main(args)\r\n  File \"C:\\Users\\doberkofler.LBITS\\AppData\\Roaming\\npm\\node_modules\\yarn\\node_modules\\node-gyp\\gyp\\pylib\\gyp\\__init__.py\", lin
e 523, in gyp_main\r\n    generator.GenerateOutput(flat_list, targets, data, params)\r\n  File \"C:\\Users\\doberkofler.LBITS\\AppData\\Roaming\\npm\\node_modul
es\\yarn\\node_modules\\node-gyp\\gyp\\pylib\\gyp\\generator\\msvs.py\", line 2003, in GenerateOutput\r\n    sln_projects, project_objects, flat=msvs_version.Fl
atSolution())\r\n  File \"C:\\Users\\doberkofler.LBITS\\AppData\\Roaming\\npm\\node_modules\\yarn\\node_modules\\node-gyp\\gyp\\pylib\\gyp\\generator\\msvs.py\"
, line 1762, in _GatherSolutionFolders\r\n    return _DictsToFolders('', root, flat)\r\n  File \"C:\\Users\\doberkofler.LBITS\\AppData\\Roaming\\npm\\node_modul
es\\yarn\\node_modules\\node-gyp\\gyp\\pylib\\gyp\\generator\\msvs.py\", line 1715, in _DictsToFolders\r\n    for folder, contents in bucket.iteritems():\r\nAtt
ributeError: 'MSVSProject' object has no attribute 'iteritems'\r\ngyp ERR! configure error \ngyp ERR! stack Error: `gyp` failed with exit code: 1\ngyp ERR! stac
k     at ChildProcess.onCpExit (C:\\Users\\doberkofler.LBITS\\AppData\\Roaming\\npm\\node_modules\\yarn\\node_modules\\node-gyp\\lib\\configure.js:305:16)\ngyp
ERR! stack     at emitTwo (events.js:106:13)\ngyp ERR! stack     at ChildProcess.emit (events.js:191:7)\ngyp ERR! stack     at Process.ChildProcess._handle.onex
it (internal/child_process.js:215:12)\ngyp ERR! System Windows_NT 6.1.7601\ngyp ERR! command \"C:\\\\Program Files\\\\nodejs\\\\node.exe\" \"C:\\\\Users\\\\dobe
rkofler.LBITS\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\yarn\\\\node_modules\\\\node-gyp\\\\bin\\\\node-gyp.js\" \"configure\" \"--fallback-to-build\" \"-
-module=D:\\\\MyDev\\\\ljs_app\\\\trunk\\\\periscope\\\\examples\\\\yarn\\\\hanging\\\\node_modules\\\\fsevents\\\\lib\\\\binding\\\\Release\\\\node-v48-win32-x
64\\\\fse.node\" \"--module_name=fse\" \"--module_path=D:\\\\MyDev\\\\ljs_app\\\\trunk\\\\periscope\\\\examples\\\\yarn\\\\hanging\\\\node_modules\\\\fsevents\\
\\lib\\\\binding\\\\Release\\\\node-v48-win32-x64\"\ngyp ERR! cwd D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_modules\\fsevents\ngyp ERR
! node -v v6.9.1\ngyp ERR! node-gyp -v v3.4.0\ngyp ERR! not ok \nnode-pre-gyp ERR! build error \nnode-pre-gyp ERR! stack Error: Failed to execute 'node-gyp.cmd
configure --fallback-to-build --module=D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_modules\\fsevents\\lib\\binding\\Release\\node-v48-wi
n32-x64\\fse.node --module_name=fse --module_path=D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_modules\\fsevents\\lib\\binding\\Release\\
node-v48-win32-x64' (1)\nnode-pre-gyp ERR! stack     at ChildProcess.<anonymous> (D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_modules\\f
sevents\\node_modules\\node-pre-gyp\\lib\\util\\compile.js:83:29)\nnode-pre-gyp ERR! stack     at emitTwo (events.js:106:13)\nnode-pre-gyp ERR! stack     at Chi
ldProcess.emit (events.js:191:7)\nnode-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:877:16)\nnode-pre-gyp ERR! stack     at Process.ChildProc
ess._handle.onexit (internal/child_process.js:226:5)\nnode-pre-gyp ERR! System Windows_NT 6.1.7601\nnode-pre-gyp ERR! command \"C:\\\\Program Files\\\\nodejs\\\
\node.exe\" \"D:\\\\MyDev\\\\ljs_app\\\\trunk\\\\periscope\\\\examples\\\\yarn\\\\hanging\\\\node_modules\\\\fsevents\\\\node_modules\\\\node-pre-gyp\\\\bin\\\\
node-pre-gyp\" \"install\" \"--fallback-to-build\"\nnode-pre-gyp ERR! cwd D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\yarn\\hanging\\node_modules\\fsevents\
nnode-pre-gyp ERR! node -v v6.9.1\nnode-pre-gyp ERR! node-pre-gyp -v v0.6.29\nnode-pre-gyp ERR! not ok \nFailed to execute 'node-gyp.cmd configure --fallback-to
success Saved lockfile.
Done in 116.22s.
```

**If the current behavior is a bug, please provide the steps to reproduce.**
run `yarn` using this package.json file:
```json
{
  "name": "test",
  "license": "MIT",
  "version": "1.0.0",
  "repository": {},
  "scripts": {
  },
  "dependencies": {
    "babel-polyfill": "6.16.0",
    "chart.js": "2.4.0",
    "dropzone": "4.3.0",
    "floatthead": "1.4.5",
    "fullcalendar": "3.0.1",
    "jgrowl": "1.4.5",
    "jquery": "2.1.4",
    "jquery-form": "^3.50.0",
    "js-cookie": "2.1.3",
    "moment": "2.17.0",
    "mousetrap": "1.6.0",
    "mresize": "1.0.1",
    "sparklines": "1.0.0",
    "underscore": "1.8.3"
  },
  "devDependencies": {
    "autoprefixer": "~6.5.3",
    "babel-core": "~6.18.2",
    "babel-loader": "~6.2.8",
    "babel-plugin-transform-async-to-generator": "~6.16.0",
    "babel-preset-es2015": "~6.18.0",
    "bell-on-bundler-error-plugin": "~1.0.8",
    "chai": "~3.5.0",
    "chai-datetime": "~1.4.1",
    "chromedriver": "~2.25.1",
    "colors": "~1.1.2",
    "css-loader": "~0.26.0",
    "cssnano": "~3.8.1",
    "debug": "~2.3.3",
    "del": "~2.2.2",
    "es6-promise": "~4.0.5",
    "eslint": "~3.11.1",
    "eslint-loader": "~1.6.1",
    "eslint-plugin-disallow-methods": "~0.1.0",
    "exports-loader": "~0.6.3",
    "expose-loader": "~0.7.1",
    "express": "~4.14.0",
    "extract-text-webpack-plugin": "~1.0.1",
    "file-loader": "~0.9.0",
    "flow-bin": "~0.36.0",
    "fs-extra": "~1.0.0",
    "glob": "~7.1.0",
    "gulp": "~3.9.1",
    "gulp-add-src": "~0.2.0",
    "gulp-babel": "~6.1.2",
    "gulp-chmod": "~2.0.0",
    "gulp-concat": "~2.6.1",
    "gulp-debug": "~3.0.0",
    "gulp-eslint": "~3.0.1",
    "gulp-header": "~1.8.8",
    "gulp-if": "~2.0.2",
    "gulp-imagemin": "~3.1.1",
    "gulp-mocha": "~3.0.1",
    "gulp-postcss": "~6.2.0",
    "gulp-preprocess": "~2.0.0",
    "gulp-rename": "~1.2.2",
    "gulp-stylus": "~2.6.0",
    "gulp-uglify": "~2.0.0",
    "gulp-util": "~3.0.6",
    "gulp-zip": "~3.2.0",
    "html-loader": "~0.4.4",
    "image-webpack-loader": "~3.0.0",
    "imports-loader": "~0.6.5",
    "jsdoc": "~3.4.3",
    "jsonlint": "~1.6.2",
    "karma": "~1.3.0",
    "karma-chai": "~0.1.0",
    "karma-chai-datetime": "~0.1.1",
    "karma-chrome-launcher": "~2.0.0",
    "karma-firefox-launcher": "~1.0.0",
    "karma-ie-launcher": "~1.0.0",
    "karma-mocha": "~1.3.0",
    "karma-mocha-reporter": "~2.2.1",
    "karma-phantomjs-launcher": "~1.0.2",
    "karma-sinon": "~1.0.5",
    "karma-sourcemap-loader": "~0.3.7",
    "karma-webpack": "~1.8.0",
    "map-stream": "~0.0.6",
    "mkdirp": "~0.5.1",
    "mocha": "~3.2.0",
    "ms": "~0.7.2",
    "null-loader": "~0.1.1",
    "on-build-webpack": "~0.1.0",
    "postcss-loader": "~1.1.1",
    "preprocess-loader": "~0.2.0",
    "run-sequence": "~1.2.2",
    "selenium-webdriver": "~3.0.1",
    "sinon": "~1.17.6",
    "strip-json-comments": "~2.0.1",
    "style-loader": "~0.13.1",
    "stylelint": "~7.5.0",
    "stylelint-config-standard": "~14.0.0",
    "stylus-loader": "~2.4.0",
    "through": "~2.3.8",
    "url-loader": "~0.5.7",
    "webpack": "~1.13.3",
    "webpack-bundle-analyzer": "^2.1.1",
    "webpack-validator": "~2.2.9",
    "yargs": "~6.4.0"
  }
}
```

**What is the expected behavior?**

**Please mention your node.js, yarn and operating system version.**
node: 6.9.1
yarn: 0.17.10
os: windows 7

**Comment by zaggy** [comment]
You can safely ignore this error as yarn is just trying to install fsevents which is an optional dependency of whatever package you have in dependencies (probably gulp) and it is not available for win32 platform. YOu may want to run the command with --ignore-optional key to produce cleaner output:

λ yarn --ignore-optional
yarn install v0.17.10
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.0.15: The platform "win32" is incompatible with this module.
info "fsevents@1.0.15" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 29.52s.

There is a duplicate issue on optional dependencies which is closed but obviously the issue still persists: #1059

**Comment by olingern** [comment]
Possible duplicate of #2136

**Comment by voxsim** [comment]
@doberkofler do you still have this issue?

**Comment by doberkofler** [comment]
Solved in 0.23.4
